nautobot/extras/nautobot_app/__init__.py

```diff
@@ -0,0 +1,136 @@
+from django.apps import AppConfig
+from operator import getitem
+from collections import OrderedDict
+
+from nautobot.extras.plugins.utils import import_object
+from nautobot.extras.registry import registry
+from nautobot.utilities.choices import ButtonColorChoices
+
+
+class NautobotConfig(AppConfig):
+    menu_tabs = "navigation.menu_tabs"
+
+    def ready(self):
+        if self.__module__.endswith(".apps"):
+            self.__module__ = self.__module__[:-5]
+        menu_items = import_object(f"{self.__module__}.{self.menu_tabs}")
+        if menu_items is not None:
+            register_menu_items(menu_items)
+
+
+def register_menu_items(class_list):
+    for nav_tab in class_list:
+        registry["nav_menu"]["tabs"][nav_tab.name] = {
+            "weight": nav_tab.weight,
+            "groups": registry["nav_menu"]["tabs"][nav_tab.name]["groups"]
+            if registry["nav_menu"]["tabs"].get(nav_tab.name)
+            else {},
+            "permissions": [],
+        }
+
+        tab_perms = []
+        for group in nav_tab.groups:
+            registry["nav_menu"]["tabs"][nav_tab.name]["groups"][group.name] = {
+                "weight": group.weight,
+                "items": registry["nav_menu"]["tabs"][nav_tab.name]["groups"][group.name]["items"]
+                if registry["nav_menu"]["tabs"][nav_tab.name]["groups"].get(group.name)
+                else {},
+                "permissions": [],
+            }
+
+            group_perms = []
+            for item in group.items:
+                registry["nav_menu"]["tabs"][nav_tab.name]["groups"][group.name]["items"][item.link] = item
+                group_perms += item.permissions
+            registry["nav_menu"]["tabs"][nav_tab.name]["groups"][group.name]["permissions"] = group_perms
+            tab_perms += group_perms
+
+        registry["nav_menu"]["tabs"][nav_tab.name]["permissions"] += tab_perms
+        registry["nav_menu"]["tabs"][nav_tab.name]["groups"] = OrderedDict(
+            sorted(registry["nav_menu"]["tabs"][nav_tab.name]["groups"].items(), key=lambda x: getitem(x[1], "weight"))
+        )
+
+    registry["nav_menu"]["tabs"] = OrderedDict(
+        sorted(registry["nav_menu"]["tabs"].items(), key=lambda x: getitem(x[1], "weight"))
+    )
```

Comment:
I must be missing something in these diffs, because I don't see where you're actually rendering the `nav_menu` using all of this registered data?

Reply:
Have a look in `nautobot/core/templates/inc/nav_menu.html`

Reply:
Thanks. I misread the GitHub display as saying that file was deleted entirely, not that it just had a large chunk of deleted diffs. :-)